packages/elements/src/rating/index.ts

```diff
@@ -42,123 +51,248 @@ export class Rating extends BasicElement {
+   * @returns {void}
    */
-  private get maxNumber (): number {
-    let max = parseFloat(this.max);
-    if (!this.max || isNaN(max)) {
-      return 5;
+  private interactiveChanged (): void {
+    if (this.interactive) {
+      this.setAttribute('role', 'slider');
+      this.setAttribute('aria-valuemin', this.MIN_VALUE.toString());
+      this.setAttribute('aria-valuenow', this.value);
+      this.setAttribute('aria-valuemax', this.max);
+      this.setAttribute('tabindex', '0');
+    }
+    else {
+      this.removeAttribute('role');
+      this.removeAttribute('aria-valuemin');
+      this.removeAttribute('aria-valuenow');
+      this.removeAttribute('aria-valuemax');
+      this.removeAttribute('tabindex');
     }
-    // Prevent decimal max value
-    max = max % 1 === 0 ? max : Math.round(max);
-    return max;
   }
 
   /**
-   * Compute rating based on max number of stars and value.
-   * Note: to speed up the component, hover state is implemented using CSS only.
-   * CSS3 specification does not allow to select items preceding the hover, but allows to select the following items.
-   * Therefore `flex: reverse` style is applied and the items are constructed in the reverse mode to mimic the correct behaviour.
-   * @param {Number} max Number of stars
-   * @param {Number} value Value
+   * Handles key input
+   * @param event Key down event object
    * @returns {void}
    */
-  private computeRating (max: number, value: number): void {
+  protected onKeyDown (event: KeyboardEvent): void {
+    if (event.defaultPrevented || !this.interactive) {
+      return;
+    }
 
-    this.items = [];
-    for (let i = 0; i < max; i += 1) {
-      const reverseIdx = value - (max - i) + 1;
-      const v = reverseIdx > 0 ? Math.min(1, reverseIdx) : 0;
+    // Ignore special keys
+    if (event.shiftKey || event.ctrlKey || event.altKey || event.metaKey) {
+      return;
+    }
 
-      const selected = v >= 0.75 ? 'full' : v >= 0.25 ? 'half' : false;
-      this.items.push({ item: selected ? `icon icon-${selected}` : 'icon' });
+    switch (event.key) {
+      case 'Right':
+      case 'Up':
+      case 'ArrowRight':
+      case 'ArrowUp':
+        this.stepUp();
+        break;
+      case 'Left':
+      case 'Down':
+      case 'ArrowLeft':
+      case 'ArrowDown':
+        this.stepDown();
+        break;
+      case 'Home':
+        this.stepDown(this.MIN_VALUE);
+        break;
+      case 'End':
+        this.stepUp(this.maxNumber);
+        break;
+      default:
+        return;
     }
 
-    this.valuePrevious = value;
-    this.requestUpdate();
+    event.preventDefault();
   }
 
   /**
-   * Process click event to set the new value
-   * @param {number} index index of star
+   * Update value and fired value-changed event
+   * @param value value to updated
    * @returns {void}
    */
-  private handleTap (index: number): void {
-    if (!this.interactive) {
+  private updateValue (value: string): void {
```

Comment:
```suggestion
  private notifyValueChange (value: string): void {
```
Just to be consistent with the other component.

Reply:
updated